rotkehlchen/tests/unit/test_protocol_balances.py

```diff
@@ -296,21 +302,32 @@ def test_thegraph_balances_vested_arbitrum_one(
                 tx_hash,
             ),
         )
-    wait_until_all_nodes_connected(
-        connect_at_start=arbitrum_one_manager_connect_at_start,
-        evm_inquirer=arbitrum_one_inquirer,
-    )
-    thegraph_balances_inquirer = ThegraphBalancesArbitrumOne(
-        database=arbitrum_one_transaction_decoder.database,
-        evm_inquirer=arbitrum_one_inquirer,
-    )
-    thegraph_balances = thegraph_balances_inquirer.query_balances()
-    asset = A_GRT_ARB.resolve_to_evm_token()
-    for balance, address in zip(expected_grt_balances, arbitrum_one_accounts, strict=True):
-        assert thegraph_balances[address].assets[asset] == Balance(
-            amount=balance,
-            usd_value=balance * FVal(1.5),
+    with patch(
```

Comment:
Okay so this works, but has two problems.

1. You absolutely need to get into the habit of explaining things that future readers will have no idea about. Future mohit or future lefteris coming in here and looking at this patching will wonder wtf? And why? Though you have a good reason to do it. Explain right above with a comment that you patch it to guarantee order of returned list and thus making the test's remote calls deterministic for VCR
2. The patch works ... but makes it so that it does not test the `process_staking_events` function. What I woudl do instead is make a mock function, which calls the original function and then takes the return value and orders it in some order. Let;s say alphabetical of the tuple's hash or something.